### PR TITLE
[chore] Update contributing guidelines

### LINKED ISSUE #21: Getting 404 for some guthub urls
For some reason I cant access:
https://github.com/orgs/open-telemetry/teams/go-instrumentation-approvers
https://github.com/orgs/open-telemetry/teams/go-instrumentaiton-maintainers
getting 404

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Updates the CONTRIBUTING.md to with the following, using [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md) as a template.
- Update links to community calendar, SIG notes doc and Slack channel
- Updates PR section to better describe fork and clone steps, approval workflow and link to approvers/maintainers
- Add Approvers and Maintainers section with current members

Closes the following issue:
- Closes #21

### CODE REVIEW

**Comment by MikeGoldsmith** [comment]
Blocked by:
- #135

**Comment by MikeGoldsmith** [comment]
Related to the following PR to update maintainers link:
- #133

**Review by edeNFed — APPROVED** [review]

File: CONTRIBUTING.md
Comment (MikeGoldsmith) [reply CONTRIBUTING.md:128]: Yeah, agreed - I used most of the wording from Go SDK but agree a faster cycle is better for now 👍🏻

**Review by MrAlias — APPROVED** [review]

**Pull request merged**
